src/Datadog.Trace.ClrProfiler.Managed/Integrations/Caching/Interception.cs

```diff
@@ -23,20 +24,24 @@ internal static Type[] TypeArray(params object[] objectsToCheck)
             return types;
         }
 
-        internal static string MethodKey(Type[] genericTypes, Type[] parameterTypes)
+        internal static string MethodKey(
+            Type owningType,
+            Type returnType,
+            Type[] genericTypes,
+            Type[] parameterTypes)
         {
-            var key = "m";
+            var key = $"{owningType?.AssemblyQualifiedName}_m_r{returnType.AssemblyQualifiedName}";
 
-            for (int i = 0; i < (genericTypes?.Length ?? 0); i++)
+            for (ushort i = 0; i < (genericTypes?.Length ?? 0); i++)
```

Comment:
In the future, let's keep using `int` as the default integral type in C#. There's no benefit in changing this to `ushort`, and most modern CPUs are optimized to work with full 32-bit integers. [More details.](https://stackoverflow.com/a/1148505/24231)